src/acl.c

```diff
@@ -1564,6 +1579,12 @@ static int ACLSelectorCheckKey(aclSelector *selector, const char *key, int keyle
         keyPattern *pattern = listNodeValue(ln);
         if ((pattern->flags & key_flags) != key_flags) continue;
         size_t plen = sdslen(pattern->pattern);
+        if (is_prefix) {
+            if (prefixmatchlen(pattern->pattern, plen, key, keylen, 0))
+                return ACL_OK;
+            else
+                continue;
+        }
```

Comment:
I think there is much shared logic that makes it reasonable to add is_prefix as argument.
But maybe it will be clearer if we turn this into a static internal function and introduce external ACLSelectorCheckKey ACLSelectorCheckKeyPrefix which will call the internal one?